testsuites/CBLTester/CBL_Functional_tests/TestSetup_FunctionalTests/test_replication.py

```diff
@@ -3215,6 +3215,260 @@ def test_replication_push_replication_guest_enabled(params_from_base_test_setup,
     replicator.stop(repl)
 
 
+@pytest.mark.listener
+@pytest.mark.replication
+def test_doc_removal_from_channel(params_from_base_test_setup):
+    """
+        @summary:
+        1. Create 2 docs in CBL with channel A, B
+        2. Create user in SGW with channel A, B.
+        3. push_pull replicate to SGW
+        4. remove doc A from channel A
+        5. Remove doc B from channel A , B
+        6. continue push_pull replication
+        7. Verify user can only access doc A, but not doc B
+
+    """
+    sg_db = "db"
+    sg_url = params_from_base_test_setup["sg_url"]
+    sg_admin_url = params_from_base_test_setup["sg_admin_url"]
+    sg_blip_url = params_from_base_test_setup["target_url"]
+    base_url = params_from_base_test_setup["base_url"]
+    cluster_config = params_from_base_test_setup["cluster_config"]
+    sg_config = params_from_base_test_setup["sg_config"]
+    cluster_config = params_from_base_test_setup["cluster_config"]
+    sg_config = params_from_base_test_setup["sg_config"]
+    db = params_from_base_test_setup["db"]
+    cbl_db = params_from_base_test_setup["source_db"]
+    sync_gateway_version = params_from_base_test_setup["sync_gateway_version"]
+
+    username = "autotest"
+    password = "password"
+    documentObj = Document(base_url)
+
+    if sync_gateway_version < "2.5.0":
+        pytest.skip('This test cannnot run with sg version below 2.5.0')
+
+    c = cluster.Cluster(config=cluster_config)
+    c.reset(sg_config_path=sg_config)
+
+    channels = ["ABC", "DEF"]
+
+    sg_client = MobileRestClient()
+    authenticator = Authenticator(base_url)
+    replicator = Replication(base_url)
+
+    # 1. Create 2 docs in CBL with channel ABC, DEF
+    cbl_ids = db.create_bulk_docs(2, "cbl", db=cbl_db, channels=channels)
+
+    # 2. Create users in SGW with channel ABC, DEF
+    sg_client.create_user(sg_admin_url, sg_db, username, password=password, channels=channels)
+    cookie, session_id = sg_client.create_session(sg_admin_url, sg_db, username)
+    session = cookie, session_id
+
+    # 3. push_pull replicate to SGW
+    replicator = Replication(base_url)
```

Comment:
duplicate initialisation of replicator object

Reply:
done